Firestore/Source/Public/FIRWriteBatch.h

```diff
@@ -98,7 +98,8 @@ FIR_SWIFT_NAME(WriteBatch)
  * Commits all of the writes in this write batch as a single atomic unit.
  *
  * @param completion A block to be called once all of the writes in the batch have been
- * successfully written to the backend as an atomic unit.
+ * successfully written to the backend as an atomic unit. This block is not guaranteed
+ * to execute if the device remains offline.
```

Comment:
This doesn't directly address the concern expressed in the underlying issue and doesn't directly caution users against using it.

Better would be something like:

This block _only_ runs while the device is online and the commit has completed against the server. The completion handler will not be called while the device is offline.